multimodal/src/autogluon/multimodal/data/datamodule.py

```diff
@@ -71,13 +74,15 @@ def __init__(
         self.val_data = val_data
         self.test_data = test_data
         self.predict_data = predict_data
+        self.corpus = corpus
```

Comment:
After reading the source code, I’m not sure why do we need it.

Reply:
At least, corpus is not a good name.

Reply:
For the text-to-text matching, one document may appear in several samples. The corpus is to store the id-to-doc mapping and the train_df is only needs to contain the doc indexes. This can reduce the memory usages since we don't need to replicate one long document several times in different samples. Storing only the indexes in the train_df is more memory-efficient. Actually, the ir_dataset uses this kind of format, e.g., https://ir-datasets.com/antique.html

Reply:
`matcher_collection ` seems not to give the hint of data. How about `metadata`, `database`, or `corpus_metadata`?

Reply:
Changed the name to `id_mappings`, which is more straightforward to understand.